### ISSUE TITLE
Drifted marker positions

### ISSUE THREAD

**Issue by marko-pi** [matplotlib/matplotlib#8533]
Matplotlib draws markers centered around individual pixels of the picture, which means that actual marker coordinates are approximated.  This is a considerable problem when smaller resolution plots are required (e.g. creating videos).  In particular, markers with slightly different coordinates are displayed on the same position.

For MWE below, all six points have different ordinate values, yet it seems as if points 2,3 and points 4,5,6 have the same value.

![bad](https://cloud.githubusercontent.com/assets/18025812/25396183/ee54cdf8-29e3-11e7-8cd2-d5085592a4c0.png)

MWE:
```
import matplotlib
from matplotlib import pyplot as plt

coor = [[0.5,0.525,0.55,0.575,0.6,0.625],[0.5,0.501,0.502,0.503,0.504,0.505]]

fig = plt.figure(figsize=(3.5,3.5))
plts=fig.add_subplot(1,1,1)
fig.subplots_adjust(left=0.01, right=0.99, bottom=0.01, top=0.99, hspace=0, wspace=0)

plts.set_xlim([0,1])
plts.set_ylim([0,1])
plts.get_xaxis().set_visible(False)
plts.get_yaxis().set_visible(False)

grph = plts.scatter(coor[0],coor[1],facecolor='k',marker='o',lw=0,s=25)
fig.savefig('test.png', bbox_inches='tight', dpi=100)
```

According to my visual experience, about two intermediate steps should be included so that the problem is no longer perceived.

The problem can be worked around by creating larger resolution plots and then downscaling them, which is sometimes a tedious process with bad side effects.  

![good](https://cloud.githubusercontent.com/assets/18025812/25395890/e672ab9c-29e2-11e7-8a99-4bd8bef3560a.jpg)

This problem was also reported on Stackoverflow:

http://stackoverflow.com/questions/43589568/increase-point-drawing-precision/

Best regards

**Comment by phobson** [comment]
@marko-pi could you summarize the stackoverflow question here in case the link goes bad?

Also, could you make more clear what your proposed solution is and provide a more descriptive title for the issue?

**Comment by tacaswell** [comment]
This is a duplicate of https://github.com/matplotlib/matplotlib/issues/7233 which is due to an underlying optimization ('stamping' the markers).

**Comment by marko-pi** [comment]
@tacaswell I agree, this is the marker stamping problem.
Is it still necessary to summarise the stackoverflow question?
Will the second report of the same problem make it more urgent?

**Comment by tacaswell** [comment]
I am unclear if this is a regression or an issue that we have always had an no one noticed.

**Comment by anntzer** [comment]
It is a regression between 1.3 and 1.4 (https://github.com/matplotlib/matplotlib/issues/7262#issuecomment-253357141).
